### PR TITLE
Add patches only to files having error in scala 3

### PR DESCRIPTION
**Pull request by mlachkar**
To avoid computing unecessary patches

Still need to improve logging for this commit.

### CODE REVIEW

**Comment by adpi2** [comment]
Interesting idea! Also if there is some compiler errors you can compute the patches only in the erroneous files and not the other ones.

**Comment by mlachkar** [comment]
I do like this idea.

File: interfaces/compiler/src/main/java/compiler/interfaces/QuietReporter.java [hunk 1]
```diff
@@ -3,7 +3,10 @@
 import dotty.tools.dotc.core.Contexts.Context;
 import dotty.tools.dotc.reporting.Diagnostic;
 import dotty.tools.dotc.reporting.Reporter;
+import java.util.*;
 
 public class QuietReporter extends Reporter {
-  public void doReport(Diagnostic d, Context ctx) {}
+  HashSet<String> filesWithErrors = new HashSet<String>();
```
Comment (adpi2) [inline interfaces/compiler/src/main/java/compiler/interfaces/QuietReporter.java:9]: Not used

File: project/project/metals.sbt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// DO NOT EDIT! This file is auto-generated.
+// This file makes sure that the JDI tools are in the sbt classpath.
+// JDI tools are used by the debug adapter server.
+
+addSbtPlugin("org.scala-debugger" % "sbt-jdi-tools" % "1.1.1")
```
Comment (adpi2) [inline project/project/metals.sbt:5]: This file should be added to .gitignore

File: migrate/src/main/scala/migrate/ScalaMigrat.scala [hunk 3]
```diff
@@ -30,13 +30,23 @@ class ScalaMigrat(scalafixSrv: ScalafixService) {
...
+                         Success(Map[AbsolutePath, FileMigrationState.FinalState]())
+                       } else {
+                         val filesWithoutErrors = scalaFiles.diff(filesWithErr)
+                         for {
+                           initialFileToMigrate <- buildMigrationFiles(filesWithErr)
+                           _ <- compileInScala3(
+                                  initialFileToMigrate,
+                                  filesWithoutErrors ++ javaFiles ++ managedSources,
+                                  compiler
+                                )
```
Comment (adpi2) [inline migrate/src/main/scala/migrate/ScalaMigrat.scala:46]: This should fix the compilation errors in the `filesWithErr` but there may be cases where the compiler then find errors in others files, in which case we can loop to add the patches of the new `filesWithErr` until there is no more error.

Can be done in a V2 after we have found a case in which it happens.

**Review by adpi2 — APPROVED** [review]

File: migrate/src/main/scala/migrate/ScalaMigrat.scala (hunk 3 above)
Comment (mlachkar) [reply migrate/src/main/scala/migrate/ScalaMigrat.scala:46]: Good idea. I have to open an issue to get track of this.

**Pull request merged**